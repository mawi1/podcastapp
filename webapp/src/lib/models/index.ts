import { type Result, err, ok } from "neverthrow";

export type PodcastBaseData = {
  id: number;
  title: string;
  latestEpisode: Date | null;
};

export type PodcastRef = {
  id: number;
  title: string;
};

export type EpisodeBaseData = {
  id: number;
  title: string | null;
  pubDate: Date;
  durationSeconds: number | null;
};

export type ResumeData = {
  currentTime: number,
  playbackRate: number,
}

export type PaginatedList<T> = {
  items: T[];
  totalItems: number;
};

export class PageNo {
  private _p: number;

  private constructor(p: number) {
    this._p = p;
  }

  public static fromUrl(url: URL): Result<PageNo, "invalid_page_no"> {
    const param = url.searchParams.get("page");
    if (param === null) {
      return ok(new PageNo(1));
    }
    const n = parseInt(param);
    if (isNaN(n) || n < 1 || n > Number.MAX_SAFE_INTEGER) {
      return err("invalid_page_no");
    } else {
      return ok(new PageNo(n));
    }
  }

  get pageNo(): number {
    return this._p;
  }
}

export class Id {
  private _id: number;

  private constructor(id: number) {
    this._id = id;
  }

  public static fromStr(idStr: string): Result<Id, "invalid_id"> {
    const n = parseInt(idStr);
    if (isNaN(n) || n < 0 || n > Number.MAX_SAFE_INTEGER) {
      return err("invalid_id");
    } else {
      return ok(new Id(n));
    }
  }

  get id(): number {
    return this._id;
  }
}

export enum InitialSymbol {
  A = "A",
  B = "B",
  C = "C",
  D = "D",
  E = "E",
  F = "F",
  G = "G",
  H = "H",
  I = "I",
  J = "J",
  K = "K",
  L = "L",
  M = "M",
  N = "N",
  O = "O",
  P = "P",
  Q = "Q",
  R = "R",
  S = "S",
  T = "T",
  U = "U",
  V = "V",
  W = "W",
  X = "X",
  Y = "Y",
  Z = "Z",
  Numeric = "0-9",
  Other = "Other",
}

export function parseInitialSymbol(val: string): undefined | InitialSymbol {
  const match = Object.entries(InitialSymbol).find(([, value]) => value === val);
  if (match) {
    return match[1];
  } else {
    return undefined;
  }
}
