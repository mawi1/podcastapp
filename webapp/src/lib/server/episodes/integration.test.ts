import { describe, expect } from "vitest";

import { Id, PageNo } from "$lib/models";

import { dbTest, insertTestPodcasts, insertTestUser } from "../db-test";

describe("episodes", () => {
  dbTest("by id", async ({ db }) => {
    const { getEpisodeById } = await import(".");

    await insertTestUser(db);
    await insertTestPodcasts(db);

    const e = await getEpisodeById(1, Id.fromStr("343")._unsafeUnwrap());
    expect(e).toMatchSnapshot();
  });

  dbTest("by id, on playlist", async ({ db }) => {
    const { getEpisodeById } = await import(".");
    const { addToPlaylist } = await import("../playlist");

    await insertTestUser(db);
    await insertTestPodcasts(db);
    await addToPlaylist(1, 343);

    const e = await getEpisodeById(1, Id.fromStr("343")._unsafeUnwrap());
    expect(e).toMatchSnapshot();
  });

  dbTest.for([
    {
      searchParams: "?page=1",
    },
    {
      searchParams: "?page=3",
    },
    {
      searchParams: "?page=6",
    },
    {
      searchParams: "?page=7",
    },
  ])("by podcast id %s", async ({ searchParams }, { db }) => {
    const { getEpisodesByPodcastId } = await import(".");
    const { addToPlaylist } = await import("../playlist");

    await insertTestUser(db);
    await insertTestPodcasts(db);
    await addToPlaylist(1, 1469);
    await addToPlaylist(1, 1433);

    const p = await getEpisodesByPodcastId(
      1,
      Id.fromStr("10")._unsafeUnwrap(),
      PageNo.fromUrl(new URL(`http://www.example.com/podcast/${searchParams}`))._unsafeUnwrap()
    );
    expect(p).toMatchSnapshot();
  });
});
