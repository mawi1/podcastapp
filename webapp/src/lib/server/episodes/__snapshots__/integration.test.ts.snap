// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`episodes > by id 1`] = `
{
  "audioUrl": "http://podcastapp.mawi.cc/podcast/5/134/audio.mp3",
  "content": "<p>Content \${title}</p>",
  "durationSeconds": 636,
  "id": 343,
  "isOnPlaylist": false,
  "podcast": {
    "id": 5,
    "title": "Tech Tales from the Future",
  },
  "pubDate": 2024-05-05T22:00:00.040Z,
  "resumeData": null,
  "title": "Podcast (5) Episode #134",
}
`;

exports[`episodes > by id, on playlist 1`] = `
{
  "audioUrl": "http://podcastapp.mawi.cc/podcast/5/134/audio.mp3",
  "content": "<p>Content \${title}</p>",
  "durationSeconds": 636,
  "id": 343,
  "isOnPlaylist": true,
  "podcast": {
    "id": 5,
    "title": "Tech Tales from the Future",
  },
  "pubDate": 2024-05-05T22:00:00.040Z,
  "resumeData": null,
  "title": "Podcast (5) Episode #134",
}
`;

exports[`episodes > by id, with resume data 1`] = `
{
  "audioUrl": "http://podcastapp.mawi.cc/podcast/2/94/audio.mp3",
  "content": "<p>Content \${title}</p>",
  "durationSeconds": 516,
  "id": 128,
  "isOnPlaylist": false,
  "podcast": {
    "id": 2,
    "title": "Love in the Algorithm",
  },
  "pubDate": 2024-04-07T14:00:00.016Z,
  "resumeData": {
    "currentTime": 789.953,
    "playbackRate": 1.5,
  },
  "title": "Podcast (2) Episode #94",
}
`;

exports[`episodes > by podcast id { searchParams: '?page=1' } 1`] = `
{
  "items": [
    {
      "durationSeconds": 1068,
      "id": 1481,
      "isOnPlaylist": false,
      "pubDate": 2024-08-15T22:00:00.080Z,
      "title": "Podcast (10) Episode #278",
    },
    {
      "durationSeconds": 1065,
      "id": 1480,
      "isOnPlaylist": false,
      "pubDate": 2024-08-15T05:00:00.080Z,
      "title": "Podcast (10) Episode #277",
    },
    {
      "durationSeconds": 1062,
      "id": 1479,
      "isOnPlaylist": false,
      "pubDate": 2024-08-14T12:00:00.080Z,
      "title": "Podcast (10) Episode #276",
    },
    {
      "durationSeconds": 1059,
      "id": 1478,
      "isOnPlaylist": false,
      "pubDate": 2024-08-13T19:00:00.080Z,
      "title": "Podcast (10) Episode #275",
    },
    {
      "durationSeconds": 1056,
      "id": 1477,
      "isOnPlaylist": false,
      "pubDate": 2024-08-13T02:00:00.080Z,
      "title": "Podcast (10) Episode #274",
    },
    {
      "durationSeconds": 1053,
      "id": 1476,
      "isOnPlaylist": false,
      "pubDate": 2024-08-12T09:00:00.080Z,
      "title": "Podcast (10) Episode #273",
    },
    {
      "durationSeconds": 1050,
      "id": 1475,
      "isOnPlaylist": false,
      "pubDate": 2024-08-11T16:00:00.080Z,
      "title": "Podcast (10) Episode #272",
    },
    {
      "durationSeconds": 1047,
      "id": 1474,
      "isOnPlaylist": false,
      "pubDate": 2024-08-10T23:00:00.080Z,
      "title": "Podcast (10) Episode #271",
    },
    {
      "durationSeconds": 1044,
      "id": 1473,
      "isOnPlaylist": false,
      "pubDate": 2024-08-10T06:00:00.080Z,
      "title": "Podcast (10) Episode #270",
    },
    {
      "durationSeconds": 1041,
      "id": 1472,
      "isOnPlaylist": false,
      "pubDate": 2024-08-09T13:00:00.080Z,
      "title": "Podcast (10) Episode #269",
    },
    {
      "durationSeconds": 1038,
      "id": 1471,
      "isOnPlaylist": false,
      "pubDate": 2024-08-08T20:00:00.080Z,
      "title": "Podcast (10) Episode #268",
    },
    {
      "durationSeconds": 1035,
      "id": 1470,
      "isOnPlaylist": false,
      "pubDate": 2024-08-08T03:00:00.080Z,
      "title": "Podcast (10) Episode #267",
    },
    {
      "durationSeconds": 1032,
      "id": 1469,
      "isOnPlaylist": true,
      "pubDate": 2024-08-07T10:00:00.080Z,
      "title": "Podcast (10) Episode #266",
    },
    {
      "durationSeconds": 1029,
      "id": 1468,
      "isOnPlaylist": false,
      "pubDate": 2024-08-06T17:00:00.080Z,
      "title": "Podcast (10) Episode #265",
    },
    {
      "durationSeconds": 1026,
      "id": 1467,
      "isOnPlaylist": false,
      "pubDate": 2024-08-06T00:00:00.080Z,
      "title": "Podcast (10) Episode #264",
    },
    {
      "durationSeconds": 1023,
      "id": 1466,
      "isOnPlaylist": false,
      "pubDate": 2024-08-05T07:00:00.080Z,
      "title": "Podcast (10) Episode #263",
    },
    {
      "durationSeconds": 1020,
      "id": 1465,
      "isOnPlaylist": false,
      "pubDate": 2024-08-04T14:00:00.080Z,
      "title": "Podcast (10) Episode #262",
    },
    {
      "durationSeconds": 1017,
      "id": 1464,
      "isOnPlaylist": false,
      "pubDate": 2024-08-03T21:00:00.080Z,
      "title": "Podcast (10) Episode #261",
    },
    {
      "durationSeconds": 1014,
      "id": 1463,
      "isOnPlaylist": false,
      "pubDate": 2024-08-03T04:00:00.080Z,
      "title": "Podcast (10) Episode #260",
    },
    {
      "durationSeconds": 1011,
      "id": 1462,
      "isOnPlaylist": false,
      "pubDate": 2024-08-02T11:00:00.080Z,
      "title": "Podcast (10) Episode #259",
    },
    {
      "durationSeconds": 1008,
      "id": 1461,
      "isOnPlaylist": false,
      "pubDate": 2024-08-01T18:00:00.080Z,
      "title": "Podcast (10) Episode #258",
    },
    {
      "durationSeconds": 1005,
      "id": 1460,
      "isOnPlaylist": false,
      "pubDate": 2024-08-01T01:00:00.080Z,
      "title": "Podcast (10) Episode #257",
    },
    {
      "durationSeconds": 1002,
      "id": 1459,
      "isOnPlaylist": false,
      "pubDate": 2024-07-31T08:00:00.080Z,
      "title": "Podcast (10) Episode #256",
    },
    {
      "durationSeconds": 999,
      "id": 1458,
      "isOnPlaylist": false,
      "pubDate": 2024-07-30T15:00:00.080Z,
      "title": "Podcast (10) Episode #255",
    },
    {
      "durationSeconds": 996,
      "id": 1457,
      "isOnPlaylist": false,
      "pubDate": 2024-07-29T22:00:00.080Z,
      "title": "Podcast (10) Episode #254",
    },
    {
      "durationSeconds": 993,
      "id": 1456,
      "isOnPlaylist": false,
      "pubDate": 2024-07-29T05:00:00.080Z,
      "title": "Podcast (10) Episode #253",
    },
    {
      "durationSeconds": 990,
      "id": 1455,
      "isOnPlaylist": false,
      "pubDate": 2024-07-28T12:00:00.080Z,
      "title": "Podcast (10) Episode #252",
    },
    {
      "durationSeconds": 987,
      "id": 1454,
      "isOnPlaylist": false,
      "pubDate": 2024-07-27T19:00:00.080Z,
      "title": "Podcast (10) Episode #251",
    },
    {
      "durationSeconds": 984,
      "id": 1453,
      "isOnPlaylist": false,
      "pubDate": 2024-07-27T02:00:00.080Z,
      "title": "Podcast (10) Episode #250",
    },
    {
      "durationSeconds": 981,
      "id": 1452,
      "isOnPlaylist": false,
      "pubDate": 2024-07-26T09:00:00.080Z,
      "title": "Podcast (10) Episode #249",
    },
    {
      "durationSeconds": 978,
      "id": 1451,
      "isOnPlaylist": false,
      "pubDate": 2024-07-25T16:00:00.080Z,
      "title": "Podcast (10) Episode #248",
    },
    {
      "durationSeconds": 975,
      "id": 1450,
      "isOnPlaylist": false,
      "pubDate": 2024-07-24T23:00:00.080Z,
      "title": "Podcast (10) Episode #247",
    },
    {
      "durationSeconds": 972,
      "id": 1449,
      "isOnPlaylist": false,
      "pubDate": 2024-07-24T06:00:00.080Z,
      "title": "Podcast (10) Episode #246",
    },
    {
      "durationSeconds": 969,
      "id": 1448,
      "isOnPlaylist": false,
      "pubDate": 2024-07-23T13:00:00.080Z,
      "title": "Podcast (10) Episode #245",
    },
    {
      "durationSeconds": 966,
      "id": 1447,
      "isOnPlaylist": false,
      "pubDate": 2024-07-22T20:00:00.080Z,
      "title": "Podcast (10) Episode #244",
    },
    {
      "durationSeconds": 963,
      "id": 1446,
      "isOnPlaylist": false,
      "pubDate": 2024-07-22T03:00:00.080Z,
      "title": "Podcast (10) Episode #243",
    },
    {
      "durationSeconds": 960,
      "id": 1445,
      "isOnPlaylist": false,
      "pubDate": 2024-07-21T10:00:00.080Z,
      "title": "Podcast (10) Episode #242",
    },
    {
      "durationSeconds": 957,
      "id": 1444,
      "isOnPlaylist": false,
      "pubDate": 2024-07-20T17:00:00.080Z,
      "title": "Podcast (10) Episode #241",
    },
    {
      "durationSeconds": 954,
      "id": 1443,
      "isOnPlaylist": false,
      "pubDate": 2024-07-20T00:00:00.080Z,
      "title": "Podcast (10) Episode #240",
    },
    {
      "durationSeconds": 951,
      "id": 1442,
      "isOnPlaylist": false,
      "pubDate": 2024-07-19T07:00:00.080Z,
      "title": "Podcast (10) Episode #239",
    },
    {
      "durationSeconds": 948,
      "id": 1441,
      "isOnPlaylist": false,
      "pubDate": 2024-07-18T14:00:00.080Z,
      "title": "Podcast (10) Episode #238",
    },
    {
      "durationSeconds": 945,
      "id": 1440,
      "isOnPlaylist": false,
      "pubDate": 2024-07-17T21:00:00.080Z,
      "title": "Podcast (10) Episode #237",
    },
    {
      "durationSeconds": 942,
      "id": 1439,
      "isOnPlaylist": false,
      "pubDate": 2024-07-17T04:00:00.080Z,
      "title": "Podcast (10) Episode #236",
    },
    {
      "durationSeconds": 939,
      "id": 1438,
      "isOnPlaylist": false,
      "pubDate": 2024-07-16T11:00:00.080Z,
      "title": "Podcast (10) Episode #235",
    },
    {
      "durationSeconds": 936,
      "id": 1437,
      "isOnPlaylist": false,
      "pubDate": 2024-07-15T18:00:00.080Z,
      "title": "Podcast (10) Episode #234",
    },
    {
      "durationSeconds": 933,
      "id": 1436,
      "isOnPlaylist": false,
      "pubDate": 2024-07-15T01:00:00.080Z,
      "title": "Podcast (10) Episode #233",
    },
    {
      "durationSeconds": 930,
      "id": 1435,
      "isOnPlaylist": false,
      "pubDate": 2024-07-14T08:00:00.080Z,
      "title": "Podcast (10) Episode #232",
    },
    {
      "durationSeconds": 927,
      "id": 1434,
      "isOnPlaylist": false,
      "pubDate": 2024-07-13T15:00:00.080Z,
      "title": "Podcast (10) Episode #231",
    },
    {
      "durationSeconds": 924,
      "id": 1433,
      "isOnPlaylist": true,
      "pubDate": 2024-07-12T22:00:00.080Z,
      "title": "Podcast (10) Episode #230",
    },
    {
      "durationSeconds": 921,
      "id": 1432,
      "isOnPlaylist": false,
      "pubDate": 2024-07-12T05:00:00.080Z,
      "title": "Podcast (10) Episode #229",
    },
  ],
  "totalItems": 278,
}
`;

exports[`episodes > by podcast id { searchParams: '?page=3' } 1`] = `
{
  "items": [
    {
      "durationSeconds": 768,
      "id": 1381,
      "isOnPlaylist": false,
      "pubDate": 2024-06-06T02:00:00.080Z,
      "title": "Podcast (10) Episode #178",
    },
    {
      "durationSeconds": 765,
      "id": 1380,
      "isOnPlaylist": false,
      "pubDate": 2024-06-05T09:00:00.080Z,
      "title": "Podcast (10) Episode #177",
    },
    {
      "durationSeconds": 762,
      "id": 1379,
      "isOnPlaylist": false,
      "pubDate": 2024-06-04T16:00:00.080Z,
      "title": "Podcast (10) Episode #176",
    },
    {
      "durationSeconds": 759,
      "id": 1378,
      "isOnPlaylist": false,
      "pubDate": 2024-06-03T23:00:00.080Z,
      "title": "Podcast (10) Episode #175",
    },
    {
      "durationSeconds": 756,
      "id": 1377,
      "isOnPlaylist": false,
      "pubDate": 2024-06-03T06:00:00.080Z,
      "title": "Podcast (10) Episode #174",
    },
    {
      "durationSeconds": 753,
      "id": 1376,
      "isOnPlaylist": false,
      "pubDate": 2024-06-02T13:00:00.080Z,
      "title": "Podcast (10) Episode #173",
    },
    {
      "durationSeconds": 750,
      "id": 1375,
      "isOnPlaylist": false,
      "pubDate": 2024-06-01T20:00:00.080Z,
      "title": "Podcast (10) Episode #172",
    },
    {
      "durationSeconds": 747,
      "id": 1374,
      "isOnPlaylist": false,
      "pubDate": 2024-06-01T03:00:00.080Z,
      "title": "Podcast (10) Episode #171",
    },
    {
      "durationSeconds": 744,
      "id": 1373,
      "isOnPlaylist": false,
      "pubDate": 2024-05-31T10:00:00.080Z,
      "title": "Podcast (10) Episode #170",
    },
    {
      "durationSeconds": 741,
      "id": 1372,
      "isOnPlaylist": false,
      "pubDate": 2024-05-30T17:00:00.080Z,
      "title": "Podcast (10) Episode #169",
    },
    {
      "durationSeconds": 738,
      "id": 1371,
      "isOnPlaylist": false,
      "pubDate": 2024-05-30T00:00:00.080Z,
      "title": "Podcast (10) Episode #168",
    },
    {
      "durationSeconds": 735,
      "id": 1370,
      "isOnPlaylist": false,
      "pubDate": 2024-05-29T07:00:00.080Z,
      "title": "Podcast (10) Episode #167",
    },
    {
      "durationSeconds": 732,
      "id": 1369,
      "isOnPlaylist": false,
      "pubDate": 2024-05-28T14:00:00.080Z,
      "title": "Podcast (10) Episode #166",
    },
    {
      "durationSeconds": 729,
      "id": 1368,
      "isOnPlaylist": false,
      "pubDate": 2024-05-27T21:00:00.080Z,
      "title": "Podcast (10) Episode #165",
    },
    {
      "durationSeconds": 726,
      "id": 1367,
      "isOnPlaylist": false,
      "pubDate": 2024-05-27T04:00:00.080Z,
      "title": "Podcast (10) Episode #164",
    },
    {
      "durationSeconds": 723,
      "id": 1366,
      "isOnPlaylist": false,
      "pubDate": 2024-05-26T11:00:00.080Z,
      "title": "Podcast (10) Episode #163",
    },
    {
      "durationSeconds": 720,
      "id": 1365,
      "isOnPlaylist": false,
      "pubDate": 2024-05-25T18:00:00.080Z,
      "title": "Podcast (10) Episode #162",
    },
    {
      "durationSeconds": 717,
      "id": 1364,
      "isOnPlaylist": false,
      "pubDate": 2024-05-25T01:00:00.080Z,
      "title": "Podcast (10) Episode #161",
    },
    {
      "durationSeconds": 714,
      "id": 1363,
      "isOnPlaylist": false,
      "pubDate": 2024-05-24T08:00:00.080Z,
      "title": "Podcast (10) Episode #160",
    },
    {
      "durationSeconds": 711,
      "id": 1362,
      "isOnPlaylist": false,
      "pubDate": 2024-05-23T15:00:00.080Z,
      "title": "Podcast (10) Episode #159",
    },
    {
      "durationSeconds": 708,
      "id": 1361,
      "isOnPlaylist": false,
      "pubDate": 2024-05-22T22:00:00.080Z,
      "title": "Podcast (10) Episode #158",
    },
    {
      "durationSeconds": 705,
      "id": 1360,
      "isOnPlaylist": false,
      "pubDate": 2024-05-22T05:00:00.080Z,
      "title": "Podcast (10) Episode #157",
    },
    {
      "durationSeconds": 702,
      "id": 1359,
      "isOnPlaylist": false,
      "pubDate": 2024-05-21T12:00:00.080Z,
      "title": "Podcast (10) Episode #156",
    },
    {
      "durationSeconds": 699,
      "id": 1358,
      "isOnPlaylist": false,
      "pubDate": 2024-05-20T19:00:00.080Z,
      "title": "Podcast (10) Episode #155",
    },
    {
      "durationSeconds": 696,
      "id": 1357,
      "isOnPlaylist": false,
      "pubDate": 2024-05-20T02:00:00.080Z,
      "title": "Podcast (10) Episode #154",
    },
    {
      "durationSeconds": 693,
      "id": 1356,
      "isOnPlaylist": false,
      "pubDate": 2024-05-19T09:00:00.080Z,
      "title": "Podcast (10) Episode #153",
    },
    {
      "durationSeconds": 690,
      "id": 1355,
      "isOnPlaylist": false,
      "pubDate": 2024-05-18T16:00:00.080Z,
      "title": "Podcast (10) Episode #152",
    },
    {
      "durationSeconds": 687,
      "id": 1354,
      "isOnPlaylist": false,
      "pubDate": 2024-05-17T23:00:00.080Z,
      "title": "Podcast (10) Episode #151",
    },
    {
      "durationSeconds": 684,
      "id": 1353,
      "isOnPlaylist": false,
      "pubDate": 2024-05-17T06:00:00.080Z,
      "title": "Podcast (10) Episode #150",
    },
    {
      "durationSeconds": 681,
      "id": 1352,
      "isOnPlaylist": false,
      "pubDate": 2024-05-16T13:00:00.080Z,
      "title": "Podcast (10) Episode #149",
    },
    {
      "durationSeconds": 678,
      "id": 1351,
      "isOnPlaylist": false,
      "pubDate": 2024-05-15T20:00:00.080Z,
      "title": "Podcast (10) Episode #148",
    },
    {
      "durationSeconds": 675,
      "id": 1350,
      "isOnPlaylist": false,
      "pubDate": 2024-05-15T03:00:00.080Z,
      "title": "Podcast (10) Episode #147",
    },
    {
      "durationSeconds": 672,
      "id": 1349,
      "isOnPlaylist": false,
      "pubDate": 2024-05-14T10:00:00.080Z,
      "title": "Podcast (10) Episode #146",
    },
    {
      "durationSeconds": 669,
      "id": 1348,
      "isOnPlaylist": false,
      "pubDate": 2024-05-13T17:00:00.080Z,
      "title": "Podcast (10) Episode #145",
    },
    {
      "durationSeconds": 666,
      "id": 1347,
      "isOnPlaylist": false,
      "pubDate": 2024-05-13T00:00:00.080Z,
      "title": "Podcast (10) Episode #144",
    },
    {
      "durationSeconds": 663,
      "id": 1346,
      "isOnPlaylist": false,
      "pubDate": 2024-05-12T07:00:00.080Z,
      "title": "Podcast (10) Episode #143",
    },
    {
      "durationSeconds": 660,
      "id": 1345,
      "isOnPlaylist": false,
      "pubDate": 2024-05-11T14:00:00.080Z,
      "title": "Podcast (10) Episode #142",
    },
    {
      "durationSeconds": 657,
      "id": 1344,
      "isOnPlaylist": false,
      "pubDate": 2024-05-10T21:00:00.080Z,
      "title": "Podcast (10) Episode #141",
    },
    {
      "durationSeconds": 654,
      "id": 1343,
      "isOnPlaylist": false,
      "pubDate": 2024-05-10T04:00:00.080Z,
      "title": "Podcast (10) Episode #140",
    },
    {
      "durationSeconds": 651,
      "id": 1342,
      "isOnPlaylist": false,
      "pubDate": 2024-05-09T11:00:00.080Z,
      "title": "Podcast (10) Episode #139",
    },
    {
      "durationSeconds": 648,
      "id": 1341,
      "isOnPlaylist": false,
      "pubDate": 2024-05-08T18:00:00.080Z,
      "title": "Podcast (10) Episode #138",
    },
    {
      "durationSeconds": 645,
      "id": 1340,
      "isOnPlaylist": false,
      "pubDate": 2024-05-08T01:00:00.080Z,
      "title": "Podcast (10) Episode #137",
    },
    {
      "durationSeconds": 642,
      "id": 1339,
      "isOnPlaylist": false,
      "pubDate": 2024-05-07T08:00:00.080Z,
      "title": "Podcast (10) Episode #136",
    },
    {
      "durationSeconds": 639,
      "id": 1338,
      "isOnPlaylist": false,
      "pubDate": 2024-05-06T15:00:00.080Z,
      "title": "Podcast (10) Episode #135",
    },
    {
      "durationSeconds": 636,
      "id": 1337,
      "isOnPlaylist": false,
      "pubDate": 2024-05-05T22:00:00.080Z,
      "title": "Podcast (10) Episode #134",
    },
    {
      "durationSeconds": 633,
      "id": 1336,
      "isOnPlaylist": false,
      "pubDate": 2024-05-05T05:00:00.080Z,
      "title": "Podcast (10) Episode #133",
    },
    {
      "durationSeconds": 630,
      "id": 1335,
      "isOnPlaylist": false,
      "pubDate": 2024-05-04T12:00:00.080Z,
      "title": "Podcast (10) Episode #132",
    },
    {
      "durationSeconds": 627,
      "id": 1334,
      "isOnPlaylist": false,
      "pubDate": 2024-05-03T19:00:00.080Z,
      "title": "Podcast (10) Episode #131",
    },
    {
      "durationSeconds": 624,
      "id": 1333,
      "isOnPlaylist": false,
      "pubDate": 2024-05-03T02:00:00.080Z,
      "title": "Podcast (10) Episode #130",
    },
    {
      "durationSeconds": 621,
      "id": 1332,
      "isOnPlaylist": false,
      "pubDate": 2024-05-02T09:00:00.080Z,
      "title": "Podcast (10) Episode #129",
    },
  ],
  "totalItems": 278,
}
`;

exports[`episodes > by podcast id { searchParams: '?page=6' } 1`] = `
{
  "items": [
    {
      "durationSeconds": 318,
      "id": 1231,
      "isOnPlaylist": false,
      "pubDate": 2024-02-20T20:00:00.080Z,
      "title": "Podcast (10) Episode #28",
    },
    {
      "durationSeconds": 315,
      "id": 1230,
      "isOnPlaylist": false,
      "pubDate": 2024-02-20T03:00:00.080Z,
      "title": "Podcast (10) Episode #27",
    },
    {
      "durationSeconds": 312,
      "id": 1229,
      "isOnPlaylist": false,
      "pubDate": 2024-02-19T10:00:00.080Z,
      "title": "Podcast (10) Episode #26",
    },
    {
      "durationSeconds": 309,
      "id": 1228,
      "isOnPlaylist": false,
      "pubDate": 2024-02-18T17:00:00.080Z,
      "title": "Podcast (10) Episode #25",
    },
    {
      "durationSeconds": 306,
      "id": 1227,
      "isOnPlaylist": false,
      "pubDate": 2024-02-18T00:00:00.080Z,
      "title": "Podcast (10) Episode #24",
    },
    {
      "durationSeconds": 303,
      "id": 1226,
      "isOnPlaylist": false,
      "pubDate": 2024-02-17T07:00:00.080Z,
      "title": "Podcast (10) Episode #23",
    },
    {
      "durationSeconds": 300,
      "id": 1225,
      "isOnPlaylist": false,
      "pubDate": 2024-02-16T14:00:00.080Z,
      "title": "Podcast (10) Episode #22",
    },
    {
      "durationSeconds": 297,
      "id": 1224,
      "isOnPlaylist": false,
      "pubDate": 2024-02-15T21:00:00.080Z,
      "title": "Podcast (10) Episode #21",
    },
    {
      "durationSeconds": 294,
      "id": 1223,
      "isOnPlaylist": false,
      "pubDate": 2024-02-15T04:00:00.080Z,
      "title": "Podcast (10) Episode #20",
    },
    {
      "durationSeconds": 291,
      "id": 1222,
      "isOnPlaylist": false,
      "pubDate": 2024-02-14T11:00:00.080Z,
      "title": "Podcast (10) Episode #19",
    },
    {
      "durationSeconds": 288,
      "id": 1221,
      "isOnPlaylist": false,
      "pubDate": 2024-02-13T18:00:00.080Z,
      "title": "Podcast (10) Episode #18",
    },
    {
      "durationSeconds": 285,
      "id": 1220,
      "isOnPlaylist": false,
      "pubDate": 2024-02-13T01:00:00.080Z,
      "title": "Podcast (10) Episode #17",
    },
    {
      "durationSeconds": 282,
      "id": 1219,
      "isOnPlaylist": false,
      "pubDate": 2024-02-12T08:00:00.080Z,
      "title": "Podcast (10) Episode #16",
    },
    {
      "durationSeconds": 279,
      "id": 1218,
      "isOnPlaylist": false,
      "pubDate": 2024-02-11T15:00:00.080Z,
      "title": "Podcast (10) Episode #15",
    },
    {
      "durationSeconds": 276,
      "id": 1217,
      "isOnPlaylist": false,
      "pubDate": 2024-02-10T22:00:00.080Z,
      "title": "Podcast (10) Episode #14",
    },
    {
      "durationSeconds": 273,
      "id": 1216,
      "isOnPlaylist": false,
      "pubDate": 2024-02-10T05:00:00.080Z,
      "title": "Podcast (10) Episode #13",
    },
    {
      "durationSeconds": 270,
      "id": 1215,
      "isOnPlaylist": false,
      "pubDate": 2024-02-09T12:00:00.080Z,
      "title": "Podcast (10) Episode #12",
    },
    {
      "durationSeconds": 267,
      "id": 1214,
      "isOnPlaylist": false,
      "pubDate": 2024-02-08T19:00:00.080Z,
      "title": "Podcast (10) Episode #11",
    },
    {
      "durationSeconds": 264,
      "id": 1213,
      "isOnPlaylist": false,
      "pubDate": 2024-02-08T02:00:00.080Z,
      "title": "Podcast (10) Episode #10",
    },
    {
      "durationSeconds": 261,
      "id": 1212,
      "isOnPlaylist": false,
      "pubDate": 2024-02-07T09:00:00.080Z,
      "title": "Podcast (10) Episode #9",
    },
    {
      "durationSeconds": 258,
      "id": 1211,
      "isOnPlaylist": false,
      "pubDate": 2024-02-06T16:00:00.080Z,
      "title": "Podcast (10) Episode #8",
    },
    {
      "durationSeconds": 255,
      "id": 1210,
      "isOnPlaylist": false,
      "pubDate": 2024-02-05T23:00:00.080Z,
      "title": "Podcast (10) Episode #7",
    },
    {
      "durationSeconds": 252,
      "id": 1209,
      "isOnPlaylist": false,
      "pubDate": 2024-02-05T06:00:00.080Z,
      "title": "Podcast (10) Episode #6",
    },
    {
      "durationSeconds": 249,
      "id": 1208,
      "isOnPlaylist": false,
      "pubDate": 2024-02-04T13:00:00.080Z,
      "title": "Podcast (10) Episode #5",
    },
    {
      "durationSeconds": 246,
      "id": 1207,
      "isOnPlaylist": false,
      "pubDate": 2024-02-03T20:00:00.080Z,
      "title": "Podcast (10) Episode #4",
    },
    {
      "durationSeconds": 243,
      "id": 1206,
      "isOnPlaylist": false,
      "pubDate": 2024-02-03T03:00:00.080Z,
      "title": "Podcast (10) Episode #3",
    },
    {
      "durationSeconds": 240,
      "id": 1205,
      "isOnPlaylist": false,
      "pubDate": 2024-02-02T10:00:00.080Z,
      "title": "Podcast (10) Episode #2",
    },
    {
      "durationSeconds": 237,
      "id": 1204,
      "isOnPlaylist": false,
      "pubDate": 2024-02-01T17:00:00.080Z,
      "title": "Podcast (10) Episode #1",
    },
  ],
  "totalItems": 278,
}
`;

exports[`episodes > by podcast id { searchParams: '?page=7' } 1`] = `
{
  "items": [],
  "totalItems": 278,
}
`;
